pkg/cmd/pkg/package_install.go

```diff
@@ -1,26 +1,160 @@
+	"github.com/dcos/dcos-cli/pkg/prompt"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
+
+	"github.com/dcos/dcos-core-cli/pkg/cosmos"
+	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
 )
 
 func newCmdPackageInstall(ctx api.Context) *cobra.Command {
-	var appOnly, cliOnly, global, yes bool
+	var appOnly, cliOnly, yes bool
 	var appID, optionsPath, version string
 
 	cmd := &cobra.Command{
 		Use:   "install <package-name>",
 		Short: "Install a package",
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return invokePythonCLI(ctx)
+			packageName := args[0]
+
+			if cliOnly == appOnly {
+				cliOnly = true
+				appOnly = true
+			}
+
+			if optionsPath != "" {
+				_, err := os.Stat(optionsPath)
+				if err != nil {
+					return err
+				}
+			}
+
+			c, err := cosmos.NewClient(ctx, pluginutil.HTTPClient(""))
+			if err != nil {
+				return err
+			}
+
+			description, err := c.PackageDescribe(packageName, version)
+			if err != nil {
+				return err
+			}
+			pkg := description.Package
+
+			link := "https://mesosphere.com/catalog-terms-conditions/#certified-services"
+			if !pkg.Selected {
+				fmt.Fprint(ctx.Out(), "This is a Community service. "+
+					"Community services are not tested for production environments. "+
+					"There may be bugs, incomplete features, incorrect documentation, or other discrepancies.\n")
+				link = "https://mesosphere.com/catalog-terms-conditions/#community-services"
+			}
+			fmt.Fprintf(ctx.Out(), "By Deploying, you agree to the Terms and Conditions %s\n", link)
+			if appOnly && pkg.PreInstallNotes != "" {
+				fmt.Fprintf(ctx.Out(), "%s\n", pkg.PreInstallNotes)
+			}
+
+			if !yes {
+				prompter := prompt.New(ctx.Input(), ctx.Out())
+				err = prompter.Confirm("Continue installing? [yes/no] ", "Yes")
+				if err != nil {
+					return err
+				}
+			}
+
+			if appOnly && description.Package.Marathon.V2AppMustacheTemplate != "" {
+				fmt.Fprintf(ctx.Out(), "Installing Marathon app for package [%s] version [%s]\n", packageName, pkg.Version)
+				err := c.PackageInstalls(appID, packageName, pkg.Version, optionsPath)
+				if err != nil {
+					return err
+				}
+			}
+
+			if cliOnly && isEmptyCli(pkg.Resource.Cli) && !isEmptyCommand(pkg.Command) {
+				return fmt.Errorf("unable to install CLI subcommand. PIP subcommands are no longer supported" +
+					" see: https://godoc.org/github.com/dcos/client-go/dcos#CosmosPackageCommand")
+			}
+
+			if cliOnly && !isEmptyCli(pkg.Resource.Cli) {
+				fmt.Fprintf(ctx.Out(), "Installing CLI subcommand for package [%s] version [%s]\n", packageName, pkg.Version)
+				pluginInfo, err := cosmos.CLIPluginInfo(description.Package, pluginutil.HTTPClient("").BaseURL())
+				if err != nil {
+					return fmt.Errorf("cannot get plugin info: %s", err)
+				}
+
+				var checksum plugin.Checksum
+				for _, contentHash := range pluginInfo.ContentHash {
+					switch contentHash.Algo {
+					case dcos.SHA256:
```

Comment:
What's with the other cases? I think a comment explaining what's going on here would help as well.

Reply:
I'm also missing a default case here.

Reply:
We only support this algorithm at the moment (https://github.com/dcos/cosmos/blob/3109d65d2e4abb66aad9b30ca9c5cb6e90469e63/cosmos-common/src/main/resources/com/mesosphere/universe/v3/schema/repo#L60).

But that's a good point that we might want to do defensive programming here and prepare for new algorithms.

I think failing would be too aggressive, what about adding a warning if the algo is unknown and skipping checksum validation?

Reply:
I think we should fail when new algorithm appears

Reply:
Fixed